src/itree.cpp

```diff
@@ -81,12 +92,22 @@ int main(int argc, char** argv){
             return 0;
         }
 
-        matcher = std::regex(vm["pattern"].as<std::string>(),
-                             std::regex::basic | std::regex::optimize );
-        if (vm.count("ignore")) {
-            exclude_matcher = std::regex(vm["ignore"].as<std::string>(),
+        if (vm.count("pattern-regex")) {
+            matcher = std::regex(vm["pattern-regex"].as<std::string>(),
+                                 std::regex::basic | std::regex::optimize );
+        }
+        if (vm.count("pattern")) {
+            assert(is_defaulted(matcher)); // Do not use both --ignore and --ignore-regex
+            matcher = vm["pattern"].as<std::string>().c_str();
```

Comment:
validity of c_str() depends on whether string repr returned by as<std::string> is a temporary?

Reply:
fixed